From: Matthias Klose <doko@debian.org>
Date: Wed, 19 Apr 2017 12:25:35 -0700
Subject: langpack-gettext

# DP: Description: support alternative gettext tree in
# DP: /usr/share/locale-langpack; if a file is present in both trees,
# DP: prefer the newer one
# DP: Upstream status: Ubuntu-Specific
---
 Lib/gettext.py | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/Lib/gettext.py b/Lib/gettext.py
index 6b215af..6e6e7cc 100644
--- a/Lib/gettext.py
+++ b/Lib/gettext.py
@@ -491,11 +491,26 @@ def find(domain, localedir=None, languages=None, all=False):
         if lang == 'C':
             break
         mofile = os.path.join(localedir, lang, 'LC_MESSAGES', '%s.mo' % domain)
+        mofile_lp = os.path.join("/usr/share/locale-langpack", lang,
+                               'LC_MESSAGES', '%s.mo' % domain)
+
+	# first look into the standard locale dir, then into the 
+	# langpack locale dir
+
+	# standard mo file
         if os.path.exists(mofile):
             if all:
                 result.append(mofile)
             else:
                 return mofile
+        
+	# langpack mofile -> use it
+        if os.path.exists(mofile_lp): 
+            if all:
+                result.append(mofile_lp)
+            else:
+                return mofile_lp
+
     return result
 
 
